lib/core/reporters/na.js

```diff
@@ -9,7 +9,25 @@ axe.addReporter('na', function(results, options, callback) {
 	}
 
 	var out = helpers.processAggregate(results, options);
+	var orientation =
+		screen.msOrientation || (screen.orientation || screen.mozOrientation || {});
+
 	callback({
+		toolOptions: options,
+		testEngine: {
+			name: 'axe-core',
+			version: axe.version
+		},
+		testRunner: {
+			name: axe._audit.brand
+		},
+		testEnvironment: {
+			userAgent: navigator.userAgent,
+			windowWidth: window.innerWidth,
+			windowHeight: window.innerHeight,
+			orientationAngle: orientation.angle,
+			orientationType: orientation.type
+		},
```

Comment:
Instead of duplicating this code a bunch of times, it'd be better to create a helper for this in `lib/core/reporters/helpers`, so you could just do:

```js
callback({
  ...getEnvironmentData(),
  toolOptions: options,
  violations: out.violations,
  passes: out.passes,
  incomplete: out.incomplete,
  inapplicable: out.inapplicable
})
```

You could probably have the URL & timestamp returned from it too. That's probably a better place for those two than the processAggregate helper.

Reply:
@WilcoFiers please see https://github.com/dequelabs/axe-core/pull/1353#discussion_r254805702

We've discussed this already, but neither @straker nor I know where to put things like this.